watcher/src/watcher.rs

```diff
@@ -101,24 +113,28 @@ impl Watcher {
     /// Sync blocks and collect signatures (and block data, when enabled).
     ///
     /// * `start` - starting block to sync.
-    /// * `max_block_height` - the max block height to sync per archive url. If
-    ///   None, continue polling.
+    /// * `max_block_height` - max block height to sync to
+    /// * `max_blocks_per_iteration` - max blocks to check for in a sync loop
+    /// * `log_sync_failures` - log node syncing failures at error! level
     ///
     /// Returns true if syncing has reached max_block_height, false if more
     /// blocks still need to be synced.
     pub fn sync_blocks(
         &self,
         start: u64,
         max_block_height: Option<u64>,
+        max_blocks_per_iteration: Option<usize>,
```

Comment:
Optional/Equivalent: Declaring the parameter as `impl Into<Option<...>>` lets the caller omit `Some()`, i.e. call with `N`, `Some(N)`, or `None`
```suggestion
        max_block_height: impl Into<Option<u64>>,
        max_blocks_per_iteration: impl Into<Option<usize>>,
```